finetune/base_models/textcnn/featurizer.py

```diff
@@ -42,8 +48,11 @@ def textcnn_featurizer(
 
         h = tf.gather(embed_weights, X)
 
-        # keep track of the classify token
-        clf_token = encoder["_classify_"]
+        # Keep track of the classify token. Token depends on tokenizer / base model weights file
+        if "roberta" in config.base_model_path:
```

Comment:
Originally I had this code along with the `embed_weights_name` code, but if I do that I get an error saying that the encoder is `NoneType` so I moved it back here

Reply:
I'm not sure how the code in between that block and this block would ever modify the value of encoder.  Could you verify that this wasn't caused by some other change be trying to move this logic back into the first conditional block?

Reply:
I don't understand why any of the code in between would modify the encoder either. But I checked again, and get this error after moving the logic to the first conditional block. This is after loading a previously trained model and then trying to predict on it.
```
>>> model.predict(["this is some text"])
Inference: 0it [00:00, ?it/s]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/indico/finetune/finetune/target_models/classifier.py", line 73, in predict
    return super().predict(X, context=context, **kwargs)
  File "/indico/finetune/finetune/base.py", line 494, in predict
    outputs = self._predict(zipped_data, **kwargs)
  File "/indico/finetune/finetune/target_models/classifier.py", line 79, in _predict
    for _,  _, start_of_doc, end_of_doc, _, proba, _, _ in self.process_long_sequence(zipped_data, **kwargs):
  File "/indico/finetune/finetune/base.py", line 926, in process_long_sequence
    pred_iterator, chunk_alignment_iterator
  File "/indico/finetune/finetune/base.py", line 469, in <genexpr>
    pred[predict_keys[0]] if len(predict_keys) == 1 else pred
  File "/usr/local/lib/python3.6/dist-packages/tqdm-4.55.1-py3.6.egg/tqdm/std.py", line 1167, in __iter__
    for obj in iterable:
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_estimator-2.3.0-py3.6.egg/tensorflow_estimator/python/estimator/estimator.py", line 613, in predict
    self.config)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_estimator-2.3.0-py3.6.egg/tensorflow_estimator/python/estimator/estimator.py", line 1163, in _call_model_fn
    model_fn_results = self._model_fn(features=features, **kwargs)
  File "/indico/finetune/finetune/model.py", line 182, in _model_fn
    lengths=features["length"]
  File "/indico/finetune/finetune/base_models/__init__.py", line 42, in get_featurizer
    return cls.featurizer(X, encoder, config, train=train, reuse=reuse, **kwargs)
  File "/indico/finetune/finetune/base_models/textcnn/featurizer.py", line 38, in textcnn_featurizer
    clf_token = encoder["_classify_"]
  File "/indico/finetune/finetune/encoding/input_encoder.py", line 152, in __getitem__
    return self.encoder[key]
TypeError: 'NoneType' object is not subscriptable
```